### PR TITLE
Fixed ExtraData in MiningConfig #4665 

### LINKED ISSUE #4665: Add ExtraData to PostMergeBlockProducer
We should change it here: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PostMergeBlockProducer.cs#L96

1. Default ExtraData should be Nethermind https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Consensus/Producers/BlockProducerBase.cs#L319
2. Allow configuration of ExtraData in our application (maybe MergeConfig?)
3. Test on testnet validators (sepolia/goerli/ropsten). It could be tested together with: https://github.com/NethermindEth/nethermind/issues/4666

### PR DESCRIPTION
**Pull request by OlegJakushkin**
Fixes #4665 

## Changes:
- [Issue 4665](https://github.com/NethermindEth/nethermind/issues/4665) describes not configurable block extra data and also highlights a few places where `ExtraData` is hardcoded in different ways.
- As block creation is a Core functionality it touches a lot of components that provide block creation. 
- This PR provides ability to configure `ExtraData` from `IMiningConfig` and expands on preexisting ExtraData roots in `MiningConfig` 
- This PR also affects unit tests and highlights the problem of hardcoded serialized block values

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**
Tested locally with Goreli, Unit Tested locally, requires wider testing as the this is implementer's first issue

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**
All new code developed in the fixing of this issue has Unit Tests `MiningConfigTest.cs`
- [x] Yes
- [ ] No

**Comments about testing, should you have some** (optional)
Not 100% of unit tests do currently pass on stand-alone windows (yet the same amount of tests fail on this PR Unit test results as on Master nethermind brunch)

### CODE REVIEW

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.RelayBuilder.cs [hunk 1]
```diff
@@ -113,8 +115,58 @@ public virtual async Task forkchoiceUpdatedV1_should_communicate_with_boost_rela
             .WithContent("{\"timestamp\":\"0x3e8\",\"prevRandao\":\"0x0000000000000000000000000000000000000000000000000000000000000000\",\"suggestedFeeRecipient\":\"0x0000000000000000000000000000000000000000\"}")
             .Respond("application/json", "{\"timestamp\":\"0x3e9\",\"prevRandao\":\"0x03783fac2efed8fbc9ad443e592ee30e61d65f471140c10ca155e937b435b760\",\"suggestedFeeRecipient\":\"0xb7705ae4c6f81b66cdb323c65f4e8133690fc099\"}");
 
+        //TODO: think about extracting an essely serialisable class, test its serializatoin sepratly, refactor with it similar methods like the one above
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.RelayBuilder.cs:118]: Minor thing - typo: sepratly

**Review by MarekM25 — COMMENTED** [review]
Great work! :) Let's make CI green and test it on sepolia validators

**Review by LukaszRozmej — APPROVED** [review]

**Comment by OlegJakushkin** [comment]
@MarekM25 Please review the update (1 hardcoded hash line update)

**Comment by OlegJakushkin** [comment]
Samples of configuration in-use
https://sepolia.etherscan.io/block/2085685 with extra
https://sepolia.etherscan.io/block/2085705 with extra
https://sepolia.etherscan.io/block/2085714 with extra
https://sepolia.etherscan.io/block/2085765 no extra
https://sepolia.etherscan.io/block/2085782 no extra
https://sepolia.etherscan.io/block/2085783 no extra 

Shout out to the Angkor team; thanks for your help in the testing of this!

**Pull request merged**